### DISCUSSION TITLE
DynamicLinks and Defer

### DISCUSSION THREAD

**Discussion by GuyoST** [sergiodxa/remix-utils#173] [Q&A] [answered]
Hello every one,

i'm using DynamicLinks to add css dynamically and it's working fine.
I would like to know how to "defer" the dynamic css (like <link rel="stylesheet" defer ... />

I would like to do that because I notice that the pagespeed website (pagespeed.web.dev) is complaining about blocking resources, and it's almost only my CSS...

I didn't find any way to add defer in LinkDescriptor.
Is there a way to do it ?

Thank you very much !!

**Comment by sergiodxa** [comment, marked as answer]
The `<link>` tag doesn't support `defer` attribute, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attributes.

The `defer` attribute is only available on `<script>` tags.

What you can do to make the resource non-blocking is to preload it with a `<link rel="preload" as="style">` before the actual `<link rel="stylesheet">`.

**Reply by GuyoST** [reply]
Thank you !

We can see `<link defer ... />` so many times on the internet, I forgot it's actually only for script tag.
Thank you to remind me the good practices !!
